packages/@aws-cdk/aws-apprunner/lib/vpc-connector.ts

```diff
@@ -0,0 +1,135 @@
+  /**
+   * A list of IDs of security groups that App Runner should use for access to AWS resources under the specified subnets.
+   *
+   * @default - the default security group of the VPC which allows all outbound traffic.
+   */
+  readonly securityGroups?: ec2.ISecurityGroup[];
+
+  /**
+    * A list of subnets that App Runner should use when it associates the service with a custom Amazon VPC.
+    */
+  readonly subnets: ec2.ISubnet[];
+
+  /**
+    * The name for the VpcConnector.
+    *
+    * @default - a name generated by CloudFormation
+    */
+  readonly vpcConnectorName?: string;
+}
+
+/**
+ * Attributes for the App Runner VPC Connector
+ */
+export interface VpcConnectorAttributes {
+  /**
+   * The name of the VPC connector.
+   */
+  readonly vpcConnectorName: string;
+
+  /**
+   * The ARN of the VPC connector.
+   */
+  readonly vpcConnectorArn: string;
+
+  /**
+   * The revision of the VPC connector.
+   */
+  readonly vpcConnectorRevision: number;
+}
+
+/**
+ * Represents the App Runner VPC Connector.
+ */
+export interface IVpcConnector extends cdk.IResource {
+  /**
+   * The Name of the VPC connector.
+   */
+  readonly vpcConnectorName: string;
+
+  /**
+   * The ARN of the VPC connector.
+   */
+  readonly vpcConnectorArn: string;
+}
+
+/**
+ * The App Runner VPC Connector
+ */
+export class VpcConnector extends cdk.Resource {
+  /**
+   * Import from VPC connector name.
+   */
+  public static fromVpcConnectorName(scope: Construct, id: string, vpcConnectorName: string): IVpcConnector {
+    class Import extends cdk.Resource {
+      public vpcConnectorName = vpcConnectorName;
+      public vpcConnectorArn = cdk.Stack.of(this).formatArn({
+        resource: 'vpcconnector',
+        service: 'apprunner',
+        resourceName: vpcConnectorName,
+      })
+    }
+    return new Import(scope, id);
+  }
+
+  /**
+   * Import from VPC connector attributes.
+   */
+  public static fromServiceAttributes(scope: Construct, id: string, attrs: VpcConnectorAttributes): IVpcConnector {
+    const vpcConnectorArn = attrs.vpcConnectorArn;
+    const vpcConnectorName = attrs.vpcConnectorName;
+    const vpcConnectorRevision = attrs.vpcConnectorRevision;
+
+    class Import extends cdk.Resource {
+      public readonly vpcConnectorArn = vpcConnectorArn
+      public readonly vpcConnectorName = vpcConnectorName
+      public readonly vpcConnectorRevision = vpcConnectorRevision
+    }
+
+    return new Import(scope, id);
+  }
+  private readonly props: VpcConnectorProps;
+
+  /**
+    * The ARN of the VPC connector.
+    * @attribute
+    */
+  readonly vpcConnectorArn: string;
+
+  /**
+   * The revision of the VPC connector.
+   * @attribute
+   */
+  readonly vpcConnectorRevision: number;
+
+  /**
+    * The name of the VPC connector.
+    * @attribute
+    */
+  readonly vpcConnectorName: string;
+
+  public constructor(scope: Construct, id: string, props: VpcConnectorProps) {
+    super(scope, id);
+
+    this.props = props;
```

Comment:
Can we just use `props` directory instead of assigning this to a private property?